db/queries.py

```diff
@@ -3,44 +3,60 @@
 
 
 def get_messages_that_contain_word(word: str) -> list[Message]:
-    pass
+    return list(Message.objects.filter(text__icontains=word))
 
 
 def get_untitled_chats() -> list[Chat]:
-    pass
+    return list(Chat.objects.filter(title__startswith="Untitled"))
 
 
 def get_users_who_sent_messages_in_2015() -> list[str]:
-    pass
+    return list(Message.objects.filter(sent__year=2015).
+                values_list("user__first_name", "user__last_name"))
 
 
 def get_actual_chats() -> list[Chat]:
-    pass
+    return list(Chat.objects.filter(message__sent__year__gt=2020))
 
 
 def get_messages_contain_authors_first_name() -> list[Message]:
-    pass
+    return list(Message.objects.filter(text__contains=F("user__first_name")))
 
 
 def get_users_who_sent_messages_starts_with_m_or_a() -> list[User]:
-    pass
+    return list(User.objects.filter(
+        Q(message__text__istartswith="a") | Q(message__text__istartswith="m"))
+    )
 
 
 def get_delivered_or_admin_messages() -> list[Message]:
-    pass
+    return list(Message.objects.filter(
+        Q(user__username__startswith="admin") | Q(is_delivered=True))
+    )
 
 
 def get_count_messages_sent_by_first_name(first_name: str) -> int:
-    pass
+    return Message.objects.filter(user__first_name=first_name).count()
 
 
 def get_top_users_by_number_of_the_messages() -> list[User]:
-    pass
+    return list(User.objects.annotate(
+        num_messages=Count("message")).order_by("-num_messages"))[:3]
 
 
 def get_last_5_messages_dicts() -> list[dict]:
-    pass
+    return [
+        {"from": message.user.username, "text": message.text}
+        for message in Message.objects.select_related().order_by("-sent")[:5]
```

Comment:
Specify select related field

> There may be some situations where you wish to call select_related() with a lot of related objects, or where you don’t know all of the relations. In these cases it is possible to call select_related() with no arguments. This will follow all non-null foreign keys it can find - nullable foreign keys must be specified. This is not recommended in most cases as it is likely to make the underlying query more complex, and return more data, than is actually needed.